clients/client-python/gravitino/api/types.py

```diff
@@ -0,0 +1,1106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=C0302
+from __future__ import annotations
+from typing import List
+from .type import (
+    Type,
+    Name,
+    PrimitiveType,
+    IntegralType,
+    FractionType,
+    DateTimeType,
+    IntervalType,
+    ComplexType,
+)
+
+
+class Types:
+    """The helper class for Type. It contains all built-in types and provides utility methods."""
+
+    class NullType(Type):
+        """The data type representing `NULL` values."""
+
+        _instance: "NullType" = None
+
+        def __new__(cls):
+            if cls._instance is None:
+                cls._instance = super(Types.NullType, cls).__new__(cls)
+            return cls._instance
+
+        @classmethod
+        def get(cls) -> "NullType":
+            return cls()
+
+        def name(self) -> Name:
+            return Name.NULL
+
+        def simple_string(self) -> str:
+            return "null"
+
+    class BooleanType(PrimitiveType):
+        """The boolean type in Gravitino."""
+
+        _instance: "BooleanType" = None
+
+        def __new__(cls):
+            if cls._instance is None:
+                cls._instance = super(Types.BooleanType, cls).__new__(cls)
+            return cls._instance
+
+        @classmethod
+        def get(cls) -> "BooleanType":
+            return cls()
```

Comment:
should return `_instance` directly?

Reply:
`return cls()` will called `__new__()` function.